### DISCUSSION TITLE
Workload Identity Federation to auth from 3rd-party to GitHub

### DISCUSSION THREAD

**Discussion by caithlinpeters** [community/community#14508] [Other Feature Feedback, Questions, & Ideas]
We already use [WIF](https://cloud.google.com/blog/products/identity-security/enabling-keyless-authentication-from-github-actions) extensively from GitHub to Google Cloud Platform to enable our actions to initiate work on the platform. It's awesome.

But we'd also like to be able to do the inverse: we would like our Google Cloud Build service to initiate a pull from GitHub using WIF. Can GitHub set up a WIF pool to enable 3rd-party platforms to initiate secure connections to GitHub?

**Comment by j-flat** [comment]
I'm also interested in the reverse, all the docs mention only Github Actions to 3rd-party integration, but nothing about the inverse. Tehcnically this should be possible, but would be nice to have it confirmed before putting lot of hours to try to implement it in vain 😅
